tests/apps/courses/test_cms_plugins_course.py

```diff
@@ -0,0 +1,158 @@
+# -*- coding: utf-8 -*-
+"""
+Unit tests for the Course plugin and its model
+"""
+from django import forms
+from django.conf import settings
+from django.test import TestCase
+
+from cms.api import add_plugin, create_page
+
+from richie.apps.core.helpers import create_i18n_page
+from richie.apps.courses.cms_plugins import CoursePlugin
+from richie.apps.courses.factories import (
+    CourseFactory,
+    CourseRunFactory,
+    OrganizationFactory,
+    SubjectFactory,
+)
+from richie.apps.courses.models import CoursePluginModel
+
+
+class CoursePluginTestCase(TestCase):
+    """
+    Test that CoursePlugin correctly displays a Course's page placeholders content
+    """
+
+    def test_cms_plugins_course_form_page_choices(self):
+        """
+        The form to create a course plugin should only list course pages
+        in the select box.
+        """
+
+        class CoursePluginModelForm(forms.ModelForm):
+            """A form for testing the choices in the select box"""
+
+            class Meta:
+                model = CoursePluginModel
+                fields = ["page"]
+
+        course = CourseFactory()
+        other_page_title = "other page"
+        create_page(other_page_title, "richie/fullwidth.html", settings.LANGUAGE_CODE)
+        plugin_form = CoursePluginModelForm()
+        self.assertIn(course.extended_object.get_title(), plugin_form.as_table())
+        self.assertNotIn(other_page_title, plugin_form.as_table())
+
+    def test_cms_plugins_course_render(self):
```

Comment:
I think you need to have to separate tests: one for including a plugin on a draft page and one for including a plugin on a public page. See example here: https://github.com/openfun/richie/blob/master/tests/apps/courses/test_cms_plugins_subject.py

Reply:
i had a few bad test, i had to fix. It should be ready for a final review.

Reply:
i added the test for plugin on draft page, might as well rename the first one.. gimme a second

Reply:
I've renamed it as well.